homeassistant/components/sensor/yr.py

```diff
@@ -120,78 +123,87 @@ def unit_of_measurement(self):
         """Return the unit of measurement of this entity, if any."""
         return self._unit_of_measurement
 
-    def update(self):
-        """Get the latest data from yr.no and updates the states."""
-        now = dt_util.utcnow()
-        # Check if data should be updated
-        if self._update is not None and now <= self._update:
-            return
-
-        self._weather.update()
-
-        # Find sensor
-        for time_entry in self._weather.data['product']['time']:
-            valid_from = dt_util.parse_datetime(time_entry['@from'])
-            valid_to = dt_util.parse_datetime(time_entry['@to'])
-
-            loc_data = time_entry['location']
-
-            if self.type not in loc_data or now >= valid_to:
-                continue
-
-            self._update = valid_to
-
-            if self.type == 'precipitation' and valid_from < now:
-                self._state = loc_data[self.type]['@value']
-                break
-            elif self.type == 'symbol' and valid_from < now:
-                self._state = loc_data[self.type]['@number']
-                break
-            elif self.type in ('temperature', 'pressure', 'humidity',
-                               'dewpointTemperature'):
-                self._state = loc_data[self.type]['@value']
-                break
-            elif self.type in ('windSpeed', 'windGust'):
-                self._state = loc_data[self.type]['@mps']
-                break
-            elif self.type == 'windDirection':
-                self._state = float(loc_data[self.type]['@deg'])
-                break
-            elif self.type in ('fog', 'cloudiness', 'lowClouds',
-                               'mediumClouds', 'highClouds'):
-                self._state = loc_data[self.type]['@percent']
-                break
-
 
 class YrData(object):
     """Get the latest data and updates the states."""
 
-    def __init__(self, coordinates):
+    def __init__(self, hass, coordinates, devices):
         """Initialize the data object."""
         self._url = 'http://api.yr.no/weatherapi/locationforecast/1.9/?' \
             'lat={lat};lon={lon};msl={msl}'.format(**coordinates)
 
         self._nextrun = None
+        self.devices = devices
         self.data = {}
-        self.update()
+        self.hass = hass
+        async_track_time_change(self.hass, self.async_update, second=0,
+                                minute=0, hour=3)
```

Comment:
This request schedules a once per day update in case there was a connection error etc in the async_update method. It will effectively "restart" the updating in case anything went wrong.